build_tools/github_actions/ci_build_stablehlo_python_api.sh

```diff
@@ -36,7 +36,9 @@ cmake -GNinja \
   -DCMAKE_CXX_COMPILER=clang++ \
   -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
   -DCMAKE_C_COMPILER=clang \
-  -DCMAKE_C_COMPILER_LAUNCHER=ccache
+  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+  -DLLVM_ENABLE_WARNINGS=ON \
+  -DLLVM_ENABLE_WERROR=ON
```

Comment:
Let's also enable `-DLLVM_ENABLE_PEDANTIC=ON`, following up on #455? Other than that, LGTM.

Reply:
It should be automatically enabled, as the LLVM default is on:
https://llvm.org/docs/CMake.html

This build should mimic the upstream build more closely with a `WERROR` flag added. Might make more sense to remove the `ENABLE_WARNINGS` flag, since that defaults to `On` as well

Reply:
I don't have a strong opinion, but I think I have a slight preference towards explicitly enabling both `LLVM_ENABLE_WARNINGS` and `LLVM_ENABLE_PEDANTIC` here to make the intent clear and to make sure that changes to the defaults of these flags don't silently affect our CI (although the latter is probably quite unlikely, so not a strong argument).

Reply:
Sounds good, added pedantic.